tedge_modbus/operations/common.py

```diff
@@ -32,24 +33,25 @@ def resolve_target_device(
     Returns (target_device, protocol).
     """
     if ip_address:
-        ip = ip_address or "127.0.0.1"
-        protocol = "TCP"
         target_device = {
             "protocol": "TCP",
-            "ip": ip,
+            "ip": ip_address or "127.0.0.1",
```

Comment:
This isn't directly related to your change (as you just moved the existing logic), but this line is redundant, as the `ip_address` will never be false because this block is anyway behind an if statement (e.g. `if ip_address`)